src/main.rs

```diff
@@ -98,6 +105,120 @@ enum CallOrHash {
 	Hash([u8; 32]),
 }
 
+// All the info associated with a call in the forms you may need it in.
+struct CallInfo {
+	call: NetworkRuntimeCall,
+	encoded: Vec<u8>,
+	hash: [u8; 32],
+	length: u32,
+}
+
+impl CallInfo {
```

Comment:
And just incase interestedl the other idea I would would have this be more like:

```rust
struct CallInfo<T> {
    // ... same props as before minus `call` and plus:
    _marker: std::marker::PhantomData<T>
}
```

Then instead of `get_kusama_call`, `get_polkadot_call` etc.. 3 methods when I believe only one could ever be called for a given instance of this, you could have:

```rust
impl <T: Decode> CallInfo<T> {
    fn get_call(&self) -> Result<T, &'static str> {
        T::decode(&mut &*self.bytes)
    }
}
```

Another generic "rust idiom" would be to use an enum error instead of `&'static str`. The crate `thiserror` makes this easy:

```rust
#[derive(Debug, thiserror::Error)]
enum Error {
    #[error("Cannot decode call bytes: {0}")]
    DecodeError(#[from] codec::Error)
}
```

With this, and a return type of `Result<T, Error>`, a call like `T::decode(&mut &*self.bytes)?` (note the "?") would automatically return any error decoding the bytes. And then other variants can be added to the error for other cases if needed quite easily.